retrieval/train.py

```diff
@@ -0,0 +1,104 @@
+# Train a retrieval transformer with a dataset
+
+import torch
+import rich
+from rich.console import Console
+from rich.progress import (
+    Progress,
+    BarColumn,
+    SpinnerColumn,
+    TimeElapsedColumn,
+    TextColumn,
+)
+
+
+from retro_pytorch import RETRO, TrainingWrapper
+
+console = Console()
+
+# instantiate RETRO, fit it into the TrainingWrapper with correct settings
+
+retro = RETRO(
+    max_seq_len=2048,
+    enc_dim=896,
+    enc_depth=3,
+    dec_dim=768,
+    dec_depth=12,
+    dec_cross_attn_layers=(1, 3, 6, 9),
+    heads=8,
+    dim_head=64,
+    dec_attn_dropout=0.25,
+    dec_ff_dropout=0.25,
+).cuda()
+
+# mkdir ./chunks - or this will fail
+# data set must be *.md files in /home/sdake/en
+# I used: https://github.com/istio/istio.io/tree/master/content/en
+wrapper = TrainingWrapper(
+    retro=retro,
+    knn=2,
+    chunk_size=64,
+    documents_path="/home/sdake/en",
+    glob="**/*.md",
```

Comment:
Agree, lets handle config as a separate followup. The config system does not need to be elaborate, although it may be slightly involved after we add other retreival transformers, such as MemFormer.

Reply:
Agreed.